lightkurve/interact.py

```diff
@@ -53,29 +53,37 @@ def prepare_lightcurve_datasource(lc):
     -------
     lc_source : bokeh.plotting.ColumnDataSource
     """
-    # Convert time into human readable strings, breaks with NaN time
-    # See https://github.com/KeplerGO/lightkurve/issues/116
-    if (lc.time == lc.time).all():
+    # Convert time into human readable strings
+    try:
         human_time = lc.astropy_time.isot
-    else:
+    except ValueError:
         human_time = [' '] * len(lc.flux)
 
     # Convert binary quality numbers into human readable strings
     qual_strings = []
-    for bitmask in lc.quality:
-        flag_str_list = KeplerQualityFlags.decode(bitmask)
-        if len(flag_str_list) == 0:
-            qual_strings.append(' ')
-        if len(flag_str_list) == 1:
-            qual_strings.append(flag_str_list[0])
-        if len(flag_str_list) > 1:
-            qual_strings.append("; ".join(flag_str_list))
+    if (lc.mission == 'Kepler') or (lc.mission == 'K2'):
+        for bitmask in lc.quality:
+            flag_str_list = KeplerQualityFlags.decode(bitmask)
+            if len(flag_str_list) == 0:
+                qual_strings.append(' ')
+            if len(flag_str_list) == 1:
+                qual_strings.append(flag_str_list[0])
+            if len(flag_str_list) > 1:
+                qual_strings.append("; ".join(flag_str_list))
+    else:
+        qual_strings = [' '] * len(lc.flux)
+        #TODO add TESS support
```

Comment:
Let's support TESS here properly while at it; it should only involve using `TessQualityFlags` instead of `KeplerQualityFlags` above.